chroma_core/models/host.py

```diff
@@ -1785,6 +1785,20 @@ class NoNidsPresent(Exception):
     pass
 
 
+class CreateSnapshotStep(Step):
+    def run(self, kwargs):
+        self.invoke_rust_agent_expect_result(
+            kwargs["host"], "snapshot_create", [kwargs["fsname"], kwargs["name"], kwargs["comment"]]
+        )
+
+
+class DestroySnapshotStep(Step):
+    def run(self, kwargs):
+        self.invoke_rust_agent_expect_result(
+            kwargs["host"], "snapshot_destroy", [kwargs["fsname"], kwargs["name"], kwargs["force"]]
+        )
+
+
 class MountSnapshotStep(Step):
     def run(self, kwargs):
         self.invoke_rust_agent_expect_result(kwargs["host"], "snapshot_mount", [kwargs["fsname"], kwargs["name"]])
```

Comment:
https://github.com/whamcloud/integrated-manager-for-lustre/blob/2f5c61b761f26a23a9848c6cf2e15e10bff4c276/iml-wire-types/src/snapshot.rs#L41-L46